Flaky TestManager_FakeShipper test (possible race)

This test seems flaky, CI sometimes reports failure, i was able to repro locally

To see repro

```bash
cd pkg/component/runtime
clear;for run in {1..50}; do  go clean -testcache; go test -run TestManager_FakeShipper; done
```

Output:

```
{"level":"trace","time":"2022-11-15T13:11:52+01:00","message":"registering set_state action for unit"}
{"level":"trace","time":"2022-11-15T13:11:52+01:00","message":"registering send_event action for unit"}
{"level":"trace","time":"2022-11-15T13:11:52+01:00","message":"registering kill action for unit"}
{"level":"debug","state":"healthy","message":"Fake Healthy","time":"2022-11-15T13:11:52+01:00","message":"updating unit state"}
{"level":"trace","time":"2022-11-15T13:11:52+01:00","message":"registering kill action for unit"}
{"level":"debug","state":"configuring","message":"Configuring","time":"2022-11-15T13:11:52+01:00","message":"restarting shipper output"}
{"level":"debug","state":"healthy","message":"GRPC fake event pipe connected \"unix:///tmp/at-3316922035/tmp/elastic-agent-fake-shipper-default-pipe.sock\"","time":"2022-11-15T13:11:52+01:00","message":"connected to output"}
{"level":"info","server":"unix:///tmp/at-3316922035/tmp/elastic-agent-fake-shipper-default-pipe.sock","time":"2022-11-15T13:11:52+01:00","message":"starting GRPC fake shipper server"}
{"level":"trace","time":"2022-11-15T13:11:52+01:00","message":"registering kill action for unit"}
{"level":"debug","state":"healthy","message":"Healthy","time":"2022-11-15T13:11:52+01:00","message":"updating unit state"}
{"level":"trace","time":"2022-11-15T13:11:52+01:00","message":"registering record event action for unit"}
{"level":"trace","time":"2022-11-15T13:11:52+01:00","message":"registering kill action for unit"}
{"level":"debug","state":"healthy","message":"Healthy","time":"2022-11-15T13:11:52+01:00","message":"updating unit state"}
{"level":"trace","id":"292f31d4-8774-4df6-974e-4ba9b2a9a029","time":"2022-11-15T13:11:52+01:00","message":"registering record event action"}
{"level":"trace","id":"292f31d4-8774-4df6-974e-4ba9b2a9a029","time":"2022-11-15T13:11:52+01:00","message":"subscribing for an event"}
{"level":"trace","time":"2022-11-15T13:11:52+01:00","message":"executing send_event action"}
{"level":"trace","content":{"id":"292f31d4-8774-4df6-974e-4ba9b2a9a029"},"timestamp":"2022-11-15T12:11:52Z","time":"2022-11-15T13:11:53+01:00","message":"received event"}
{"level":"trace","id":"292f31d4-8774-4df6-974e-4ba9b2a9a029","time":"2022-11-15T13:11:53+01:00","message":"subscription exists for event id"}
{"level":"trace","content":{"id":"292f31d4-8774-4df6-974e-4ba9b2a9a029"},"timestamp":"2022-11-15T12:11:52Z","time":"2022-11-15T13:11:53+01:00","message":"record_event action got subscribed event"}
{"level":"trace","id":"292f31d4-8774-4df6-974e-4ba9b2a9a029","time":"2022-11-15T13:11:53+01:00","message":"unsubscribing for an event"}
{"level":"trace","log_timer":10000,"time":"2022-11-15T13:12:02+01:00","message":"trace log ticker"}
{"level":"trace","time":"2022-11-15T13:12:02+01:00","message":"executing send_event action"}
--- FAIL: TestManager_FakeShipper (11.40s)
    manager_test.go:1962: component state changed: {State:starting Message:Starting: spawned pid '1608467' Units:map[{UnitType:input UnitID:fake-input}:{State:starting Message:Starting: spawned pid '1608467' Payload:map[] unitState:0 unitMessage:Starting unitPayload:map[] configStateIdx:0 err:<nil>} {UnitType:output UnitID:fake-default}:{State:starting Message:Starting: spawned pid '1608467' Payload:map[] unitState:0 unitMessage:Starting unitPayload:map[] configStateIdx:0 err:<nil>}] VersionInfo:{Name: Version: Meta:map[]} expectedUnits:map[{UnitType:0 UnitID:fake-input}:{state:2 configStateIdx:1 config:0xc000374750 err:<nil> logLevel:4} {UnitType:1 UnitID:fake-default}:{state:2 configStateIdx:1 config:0xc000374240 err:<nil> logLevel:4}]}
    manager_test.go:1927: shipper state changed: {State:starting Message:Starting: spawned pid '1608476' Units:map[{UnitType:input UnitID:fake-default}:{State:starting Message:Starting: spawned pid '1608476' Payload:map[] unitState:0 unitMessage:Starting unitPayload:map[] configStateIdx:0 err:<nil>} {UnitType:output UnitID:fake-default}:{State:starting Message:Starting: spawned pid '1608476' Payload:map[] unitState:0 unitMessage:Starting unitPayload:map[] configStateIdx:0 err:<nil>}] VersionInfo:{Name: Version: Meta:map[]} expectedUnits:map[{UnitType:0 UnitID:fake-default}:{state:2 configStateIdx:1 config:0xc0003741b0 err:<nil> logLevel:4} {UnitType:1 UnitID:fake-default}:{state:2 configStateIdx:1 config:0xc000374900 err:<nil> logLevel:4}]}
    manager_test.go:1962: component state changed: {State:healthy Message:Healthy: communicating with pid '1608467' Units:map[{UnitType:input UnitID:fake-input}:{State:starting Message:Starting: spawned pid '1608467' Payload:map[] unitState:0 unitMessage: unitPayload:map[] configStateIdx:0 err:<nil>} {UnitType:output UnitID:fake-default}:{State:starting Message:Starting: spawned pid '1608467' Payload:map[] unitState:0 unitMessage: unitPayload:map[] configStateIdx:0 err:<nil>}] VersionInfo:{Name:fake Version:1.0 Meta:map[input:fake]} expectedUnits:map[{UnitType:0 UnitID:fake-input}:{state:2 configStateIdx:1 config:0xc000374750 err:<nil> logLevel:4} {UnitType:1 UnitID:fake-default}:{state:2 configStateIdx:1 config:0xc000374240 err:<nil> logLevel:4}]}
    manager_test.go:1962: component state changed: {State:healthy Message:Healthy: communicating with pid '1608467' Units:map[{UnitType:input UnitID:fake-input}:{State:healthy Message:Fake Healthy Payload:map[] unitState:2 unitMessage:Fake Healthy unitPayload:map[] configStateIdx:1 err:<nil>} {UnitType:output UnitID:fake-default}:{State:configuring Message:Configuring Payload:map[] unitState:1 unitMessage:Configuring unitPayload:map[] configStateIdx:1 err:<nil>}] VersionInfo:{Name:fake Version:1.0 Meta:map[input:fake]} expectedUnits:map[{UnitType:0 UnitID:fake-input}:{state:2 configStateIdx:1 config:0xc000374750 err:<nil> logLevel:4} {UnitType:1 UnitID:fake-default}:{state:2 configStateIdx:1 config:0xc000374240 err:<nil> logLevel:4}]}
    manager_test.go:1962: component state changed: {State:healthy Message:Healthy: communicating with pid '1608467' Units:map[{UnitType:input UnitID:fake-input}:{State:healthy Message:Fake Healthy Payload:map[] unitState:2 unitMessage:Fake Healthy unitPayload:map[] configStateIdx:1 err:<nil>} {UnitType:output UnitID:fake-default}:{State:healthy Message:GRPC fake event pipe connected "unix:///tmp/at-3316922035/tmp/elastic-agent-fake-shipper-default-pipe.sock" Payload:map[] unitState:2 unitMessage:GRPC fake event pipe connected "unix:///tmp/at-3316922035/tmp/elastic-agent-fake-shipper-default-pipe.sock" unitPayload:map[] configStateIdx:1 err:<nil>}] VersionInfo:{Name:fake Version:1.0 Meta:map[input:fake]} expectedUnits:map[{UnitType:0 UnitID:fake-input}:{state:2 configStateIdx:1 config:0xc000374750 err:<nil> logLevel:4} {UnitType:1 UnitID:fake-default}:{state:2 configStateIdx:1 config:0xc000374240 err:<nil> logLevel:4}]}
    manager_test.go:1927: shipper state changed: {State:healthy Message:Healthy: communicating with pid '1608476' Units:map[{UnitType:input UnitID:fake-default}:{State:starting Message:Starting: spawned pid '1608476' Payload:map[] unitState:0 unitMessage: unitPayload:map[] configStateIdx:0 err:<nil>} {UnitType:output UnitID:fake-default}:{State:starting Message:Starting: spawned pid '1608476' Payload:map[] unitState:0 unitMessage: unitPayload:map[] configStateIdx:0 err:<nil>}] VersionInfo:{Name:fake-shipper Version:1.0 Meta:map[shipper:fake-shipper]} expectedUnits:map[{UnitType:0 UnitID:fake-default}:{state:2 configStateIdx:1 config:0xc0003741b0 err:<nil> logLevel:4} {UnitType:1 UnitID:fake-default}:{state:2 configStateIdx:1 config:0xc000374900 err:<nil> logLevel:4}]}
    manager_test.go:1927: shipper state changed: {State:healthy Message:Healthy: communicating with pid '1608476' Units:map[{UnitType:input UnitID:fake-default}:{State:healthy Message:Healthy Payload:map[] unitState:2 unitMessage:Healthy unitPayload:map[] configStateIdx:1 err:<nil>} {UnitType:output UnitID:fake-default}:{State:starting Message:Starting Payload:map[] unitState:0 unitMessage:Starting unitPayload:map[] configStateIdx:1 err:<nil>}] VersionInfo:{Name:fake-shipper Version:1.0 Meta:map[shipper:fake-shipper]} expectedUnits:map[{UnitType:0 UnitID:fake-default}:{state:2 configStateIdx:1 config:0xc0003741b0 err:<nil> logLevel:4} {UnitType:1 UnitID:fake-default}:{state:2 configStateIdx:1 config:0xc000374900 err:<nil> logLevel:4}]}
    manager_test.go:2023: 
                Error Trace:    manager_test.go:2023
                Error:          Received unexpected error:
                                context deadline exceeded
                Test:           TestManager_FakeShipper
    manager_test.go:1927: shipper state changed: {State:healthy Message:Healthy: communicating with pid '1608476' Units:map[{UnitType:input UnitID:fake-default}:{State:healthy Message:Healthy Payload:map[] unitState:2 unitMessage:Healthy unitPayload:map[] configStateIdx:1 err:<nil>} {UnitType:output UnitID:fake-default}:{State:healthy Message:Healthy Payload:map[] unitState:2 unitMessage:Healthy unitPayload:map[] configStateIdx:1 err:<nil>}] VersionInfo:{Name:fake-shipper Version:1.0 Meta:map[shipper:fake-shipper]} expectedUnits:map[{UnitType:0 UnitID:fake-default}:{state:2 configStateIdx:1 config:0xc0003741b0 err:<nil> logLevel:4} {UnitType:1 UnitID:fake-default}:{state:2 configStateIdx:1 config:0xc000374900 err:<nil> logLevel:4}]}
FAIL
```

--- a/pkg/component/runtime/manager_test.go
+++ b/pkg/component/runtime/manager_test.go
@@ -2204,7 +2204,7 @@ func TestManager_FakeShipper(t *testing.T) {
 			// wait for the event on the shipper side
 			gotEvt := make(chan error)
 			go func() {
-				actionCtx, actionCancel := context.WithTimeout(context.Background(), 10*time.Second)
+				actionCtx, actionCancel := context.WithTimeout(context.Background(), 30*time.Second)
 				_, err := m.PerformAction(actionCtx, comps[1], comps[1].Units[1], "record_event", map[string]interface{}{
 					"id": eventID.String(),
 				})
@@ -2213,7 +2213,7 @@ func TestManager_FakeShipper(t *testing.T) {
 			}()
 
 			// send the fake event
-			actionCtx, actionCancel := context.WithTimeout(context.Background(), 5*time.Second)
+			actionCtx, actionCancel := context.WithTimeout(context.Background(), 15*time.Second)
 			_, err = m.PerformAction(actionCtx, comps[0], comps[0].Units[0], "send_event", map[string]interface{}{
 				"id": eventID.String(),
 			})
@@ -2322,13 +2322,13 @@ func TestManager_FakeShipper(t *testing.T) {
 		t.Fatalf("failed early: %s", err)
 	}
 
-	endTimer := time.NewTimer(30 * time.Second)
+	endTimer := time.NewTimer(2 * time.Minute)
 	defer endTimer.Stop()
 LOOP:
 	for {
 		select {
 		case <-endTimer.C:
-			t.Fatalf("timed out after 30 seconds")
+			t.Fatalf("timed out after 2 minutes")
 		case err := <-errCh:
 			require.NoError(t, err)
 		case err := <-subErrCh:

